app/controllers/merchant/items_controller.rb

```diff
@@ -5,9 +5,68 @@ def index
     @merchant_employee = User.find(current_user.id)
   end
 
+  def new
+    @item = Item.new(session.delete(:item_params))
+  end
+
+  def edit
+    @item = Item.find(params[:item_id])
+  end
+
+  def create
+    merchant = Merchant.find(current_user.merchant.id)
+    @item = merchant.items.create(item_params)
+    if @item.save
+      flash[:success] = "Added #{@item.name} to item list"
+      redirect_to "/merchant/items"
+    else
+      flash[:failure] = "All fields must be completed before submitting:"
+      flash[:error] = @item.errors.full_messages.to_sentence
+      session[:params] = item_params
+      render :new
+    end
+  end
+
+  def update
+    @item = Item.find(params[:item_id])
+    @item.update(item_params)
+    if @item.save
+      flash[:success] = "Item has been updated"
+      redirect_to "/merchant/items"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
+  def toggle_active
```

Comment:
interesting! I'm curious about this approach as opposed to implementing this functionality in the model (like Judith did with orders: https://github.com/ajtran303/monster_shop_2005/blob/master/app/models/merchant.rb)

Reply:
I think that approach would allow us to follow restful and MVC conventions!